### PR TITLE
269 web responsive pages

### PR DESCRIPTION
**Pull request by CedricCORGE**
Adapt the web app to make it responsive

[Capture vidéo du 2024-05-13 11-48-49.webm](https://github.com/H-chauvet/RISU/assets/64684672/f501832d-cf0f-4c43-8eef-e87d64b41173)
[Capture vidéo du 2024-05-13 11-55-56.webm](https://github.com/H-chauvet/RISU/assets/64684672/90d9212f-824d-4101-b307-d369d63aad0c)
[Capture vidéo du 2024-05-13 20-31-28.webm](https://github.com/H-chauvet/RISU/assets/64684672/7ae63841-229d-40a3-b79f-039e88f850b0)

### CODE REVIEW

**Review by 31Nathan — APPROVED** [review]

File: Configurateur/front/lib/screens/container-creation/maps_screen/maps_screen.dart [hunk 1]
```diff
@@ -82,7 +85,7 @@ class MapsState extends State<MapsScreen> {
 
   @override
   void initState() {
-    checkToken();
+    //checkToken();
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/maps_screen/maps_screen.dart:88]: Why is that ?

File: Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart [hunk 2]
```diff
@@ -6,12 +6,15 @@ import 'package:front/components/alert_dialog.dart';
...
 import 'package:front/components/progress_bar.dart';
+import 'package:front/services/size_service.dart';
 import 'package:front/services/storage_service.dart';
+import 'package:front/styles/globalStyle.dart';
 import 'package:go_router/go_router.dart';
 import 'package:http/http.dart' as http;
 
-import '../../network/informations.dart';
-import '../../services/http_service.dart';
+import '../../../network/informations.dart';
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart:15]: package:

File: Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart [hunk 3]
```diff
@@ -4,11 +4,14 @@ import 'package:flutter/material.dart';
...
 import 'package:front/components/recap_panel.dart';
+import 'package:front/screens/container-creation/recap_screen/recap_screen_style.dart';
+import 'package:front/services/size_service.dart';
 import 'package:front/services/theme_service.dart';
+import 'package:front/styles/globalStyle.dart';
 import 'package:go_router/go_router.dart';
 import 'package:provider/provider.dart';
 
-import '../../styles/themes.dart';
+import '../../../styles/themes.dart';
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart:14]: package:

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 import 'package:front/components/custom_app_bar.dart';
+import 'package:front/services/size_service.dart';
 import 'package:front/services/theme_service.dart';
+import 'package:front/styles/globalStyle.dart';
 import 'package:front/styles/themes.dart';
 import 'package:go_router/go_router.dart';
 import 'package:provider/provider.dart';
 
-import '../../components/progress_bar.dart';
+import '../../../components/progress_bar.dart';
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:12]: package:

**Review by 31Nathan — CHANGES_REQUESTED** [review]

File: Configurateur/front/lib/components/container.dart [hunk 5]
```diff
@@ -96,11 +99,39 @@ class ContainerCards extends StatelessWidget {
...
+                    ? Text(
+                        "Adresse : ${container.address!}",
+                        style: TextStyle(
+                          fontSize: screenFormat == ScreenFormat.desktop
+                              ? desktopFontSize
+                              : tabletFontSize,
+                        ),
+                      )
+                    : Text(
+                        "Adresse : aucune address",
```
Comment (H-chauvet) [inline Configurateur/front/lib/components/container.dart:128]: Franglais à éviter, typographie pas respectée

File: Configurateur/front/lib/components/container.dart [hunk 6]
```diff
@@ -96,11 +99,39 @@ class ContainerCards extends StatelessWidget {
...
+                    ? Text(
+                        "Ville : ${container.city!}",
+                        style: TextStyle(
+                          fontSize: screenFormat == ScreenFormat.desktop
+                              ? desktopFontSize
+                              : tabletFontSize,
+                        ),
+                      )
+                    : Text(
+                        "Ville : pas de ville associée",
```
Comment (H-chauvet) [inline Configurateur/front/lib/components/container.dart:111]: Typographie pas respectée

**Review by H-chauvet — CHANGES_REQUESTED** [review]
Changes the typo where I commented and it's gonna be good

File: Configurateur/front/lib/components/recap_panel.dart [hunk 7]
```diff
@@ -114,28 +125,43 @@ class RecapPanel extends StatelessWidget {
             mainAxisAlignment: MainAxisAlignment.spaceBetween,
             children: [
               Container(
-                constraints: const BoxConstraints(minWidth: 140, maxWidth: 140),
+                constraints: BoxConstraints(minWidth: 7.0.w, maxWidth: 7.0.w),
```
Comment (Louismaestre) [inline Configurateur/front/lib/components/recap_panel.dart:128]: Why you don't add the value of  minWidth and maxWidth in the globalStyle ?

File: Configurateur/front/lib/components/recap_panel.dart [hunk 8]
```diff
@@ -114,28 +125,43 @@ class RecapPanel extends StatelessWidget {
...
+                    color: Colors.black,
+                    fontSize: screenFormat == ScreenFormat.desktop
+                        ? desktopFontSize
+                        : tabletFontSize,
+                  ),
                 ),
               ),
               Container(
-                constraints: const BoxConstraints(minWidth: 100),
+                constraints: BoxConstraints(minWidth: 5.0.w),
```
Comment (Louismaestre) [inline Configurateur/front/lib/components/recap_panel.dart:154]: same

File: Configurateur/front/lib/components/recap_panel.dart [hunk 9]
```diff
@@ -177,35 +205,44 @@ class RecapPanel extends StatelessWidget {
                   children: [
                     Container(
                       constraints:
-                          const BoxConstraints(minWidth: 140, maxWidth: 140),
-                      child: const Padding(
-                        padding: EdgeInsets.only(left: 10),
+                          BoxConstraints(minWidth: 7.0.w, maxWidth: 7.0.w),
```
Comment (Louismaestre) [inline Configurateur/front/lib/components/recap_panel.dart:208]: same

File: Configurateur/front/lib/components/recap_panel.dart [hunk 10]
```diff
@@ -114,28 +125,43 @@ class RecapPanel extends StatelessWidget {
...
+                        ? desktopFontSize
+                        : tabletFontSize,
+                  ),
                 ),
               ),
               Container(
-                constraints: const BoxConstraints(minWidth: 100),
+                constraints: BoxConstraints(minWidth: 5.0.w),
                 child: Padding(
                   padding: const EdgeInsets.only(right: 10, bottom: 10),
```
Comment (Louismaestre) [inline Configurateur/front/lib/components/recap_panel.dart:156]: Same with right and bottom

File: Configurateur/front/lib/components/recap_panel.dart [hunk 11]
```diff
@@ -242,7 +279,12 @@ class RecapPanel extends StatelessWidget {
...
-              child: Text('Total: ${price!}€'),
+              child: Text('Total: ${price!}€',
+                  style: TextStyle(
+                    fontSize: screenFormat == ScreenFormat.desktop
+                        ? desktopFontSize
+                        : tabletFontSize,
+                  )),
             ),
             const SizedBox(
               height: 10,
```
Comment (Louismaestre) [inline Configurateur/front/lib/components/recap_panel.dart:290]: same with height

File: Configurateur/front/lib/screens/contact/contact.dart [hunk 12]
```diff
@@ -96,8 +102,8 @@ class _ContactPageState extends State<ContactPage> {
             ),
           ),
           const SizedBox(height: 100),
-          Padding(
-            padding: const EdgeInsets.only(right: 400, left: 400),
+          FractionallySizedBox(
+            widthFactor: 0.8,
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/contact/contact.dart:106]: why you don't add de widthFactor in the globalStyle or is own style file like the others ?

File: Configurateur/front/lib/screens/container-creation/confirmation_screen/confirmation_screen.dart [hunk 13]
```diff
@@ -16,20 +20,27 @@ class ConfirmationScreen extends StatefulWidget {
...
           "Confirmation de commande",
           context: context,
         ),
         body: Center(
             child: FractionallySizedBox(
-                widthFactor: 0.3,
+                widthFactor: screenFormat == ScreenFormat.desktop
+                    ? desktopWidthFactor
+                    : tabletWidthFactor,
                 heightFactor: 0.7,
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/confirmation_screen/confirmation_screen.dart:34]: same

File: Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart [hunk 14]
```diff
@@ -782,9 +808,9 @@ class ContainerCreationState extends State<ContainerCreation> {
                     const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart:808]: Maybe you can put the value of sizebox in the globalStyle and add it to all of the others

File: Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart [hunk 15]
```diff
@@ -832,13 +874,17 @@ class ContainerCreationState extends State<ContainerCreation> {
                   ],
                 ),
               ),
+              const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart:877]: same

File: Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart [hunk 16]
```diff
@@ -849,9 +895,14 @@ class ContainerCreationState extends State<ContainerCreation> {
                   ),
                 ],
               ),
+              const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart:898]: same

File: Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart [hunk 17]
```diff
@@ -149,22 +154,32 @@ class _PaymentScreenState extends State<PaymentScreen> {
         body: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
           Center(
             child: FractionallySizedBox(
-              widthFactor: 0.6,
+              widthFactor: screenFormat == ScreenFormat.desktop
+                  ? desktopWidthFactor
+                  : tabletWidthFactor,
               child: Column(
                 children: [
                   const SizedBox(height: 50),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart:162]: same

File: Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart [hunk 18]
```diff
@@ -149,22 +154,32 @@ class _PaymentScreenState extends State<PaymentScreen> {
...
+                  Text(
                     "Coordonnées bancaires",
-                    style: TextStyle(fontSize: 17, fontWeight: FontWeight.bold),
+                    style: TextStyle(
+                        fontSize: screenFormat == ScreenFormat.desktop
+                            ? desktopFontSize
+                            : tabletFontSize,
+                        fontWeight: FontWeight.bold),
                   ),
                   const SizedBox(height: 20),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart:171]: same

File: Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart [hunk 19]
```diff
@@ -149,22 +154,32 @@ class _PaymentScreenState extends State<PaymentScreen> {
...
                   CardField(
                     controller: controller,
                   ),
                   const SizedBox(height: 100),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart:175]: same

File: Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart [hunk 20]
```diff
@@ -113,16 +117,22 @@ class RecapScreenState extends State<RecapScreen> {
                 const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart:117]: same

File: Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart [hunk 21]
```diff
@@ -113,16 +117,22 @@ class RecapScreenState extends State<RecapScreen> {
...
+                Text(
                   "Récapitulatif de la commande",
-                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
+                  style: TextStyle(
+                      fontSize: screenFormat == ScreenFormat.desktop
+                          ? desktopFontSize
+                          : tabletFontSize,
+                      fontWeight: FontWeight.bold),
                 ),
                 const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart:128]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 22]
```diff
@@ -105,7 +111,14 @@ class ShapeScreenState extends State<ShapeScreen> {
...
+            child: Text(
+              "Supprimer",
+              style: TextStyle(
+                fontSize: screenFormat == ScreenFormat.desktop
+                    ? desktopFontSize
+                    : tabletFontSize,
+              ),
+            ),
           ),
           const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:123]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 23]
```diff
@@ -333,7 +364,14 @@ class ShapeScreenState extends State<ShapeScreen> {
...
+                      child: Text(
+                        row.toString(),
+                        style: TextStyle(
+                          fontSize: screenFormat == ScreenFormat.desktop
+                              ? desktopFontSize
+                              : tabletFontSize,
+                        ),
+                      ),
                     ),
                     const SizedBox(width: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:376]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 24]
```diff
@@ -357,7 +395,14 @@ class ShapeScreenState extends State<ShapeScreen> {
                 const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:395]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 25]
```diff
@@ -357,7 +395,14 @@ class ShapeScreenState extends State<ShapeScreen> {
...
-                const Text('Nombres de colonnes'),
+                Text(
+                  'Nombres de colonnes',
+                  style: TextStyle(
+                    fontSize: screenFormat == ScreenFormat.desktop
+                        ? desktopFontSize
+                        : tabletFontSize,
+                  ),
+                ),
                 const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:406]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 26]
```diff
@@ -390,7 +435,14 @@ class ShapeScreenState extends State<ShapeScreen> {
...
+                      child: Text(
+                        column.toString(),
+                        style: TextStyle(
+                          fontSize: screenFormat == ScreenFormat.desktop
+                              ? desktopFontSize
+                              : tabletFontSize,
+                        ),
+                      ),
                     ),
                     const SizedBox(width: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:447]: same

File: Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart [hunk 27]
```diff
@@ -421,7 +473,7 @@ class ShapeScreenState extends State<ShapeScreen> {
                   Row(
                     mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                     crossAxisAlignment: CrossAxisAlignment.center,
-                    children: removeButtons(),
+                    children: removeButtons(screenFormat),
                   ),
                   const SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-creation/shape_screen/shape_screen.dart:478]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 28]
```diff
@@ -259,14 +261,22 @@ class _ContainerPageState extends State<ContainerPage> {
...
+                      : tabletFontSize,
+                ),
+              ),
               content: Container(
-                height: 220.0,
                 child: Column(
                   crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
                   children: [
                     const SizedBox(height: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:281]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 29]
```diff
@@ -277,7 +287,14 @@ class _ContainerPageState extends State<ContainerPage> {
...
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:287]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 30]
```diff
@@ -303,6 +320,7 @@ class _ContainerPageState extends State<ContainerPage> {
                           labelText: "Nouvelle description",
                           hintText: initialDesc),
                     ),
+                    const SizedBox(height: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:323]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 31]
```diff
@@ -412,18 +444,26 @@ class _ContainerPageState extends State<ContainerPage> {
...
+                      : tabletBigFontSize,
+                ),
+              ),
               content: Container(
-                height: 290.0,
                 child: Column(
                   crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
                   children: [
                     const SizedBox(height: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:468]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 32]
```diff
@@ -434,7 +474,14 @@ class _ContainerPageState extends State<ContainerPage> {
...
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:474]: same

File: Configurateur/front/lib/screens/container-list/container_list.dart [hunk 33]
```diff
@@ -476,6 +523,7 @@ class _ContainerPageState extends State<ContainerPage> {
                       decoration: const InputDecoration(
                           labelText: "Description de l'objet"),
                     ),
+                    const SizedBox(height: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/container-list/container_list.dart:526]: same

File: Configurateur/front/lib/screens/login/login.dart [hunk 34]
```diff
@@ -132,23 +137,32 @@ class LoginScreenState extends State<LoginScreen> {
                       onTap: () {
                         context.go("/password-recuperation");
                       },
-                      child: const Padding(
-                        padding: EdgeInsets.all(10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/login/login.dart:136]: maybe you can add the value of the padding in style file

File: Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart [hunk 35]
```diff
@@ -0,0 +1,227 @@
...
+        child: FractionallySizedBox(
+          widthFactor: screenFormat == ScreenFormat.desktop
+              ? desktopWidthFactor
+              : tabletWidthFactor,
+          heightFactor: 0.7,
+          child: Form(
+            key: formKey,
+            child: Column(
+              children: <Widget>[
+                const SizedBox(height: 10),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart:63]: Add in gobal style

File: Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart [hunk 36]
```diff
@@ -0,0 +1,227 @@
...
+                    },
+                    validator: (String? value) {
+                      if (value == null || value.isEmpty) {
+                        return 'Veuillez remplir ce champ';
+                      }
+                      return null;
+                    },
+                  ),
+                ),
+                const SizedBox(height: 20),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart:85]: same

File: Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart [hunk 37]
```diff
@@ -0,0 +1,227 @@
...
+                        width: screenFormat == ScreenFormat.desktop
+                            ? desktopCaptchaWidth
+                            : tabletCaptchaWidth,
+                        backgroundColor: Colors.grey[100]!,
+                        chars: configFormData.chars,
+                        length: configFormData.length,
+                        caseSensitive: true,
+                        codeExpireAfter: const Duration(minutes: 10),
+                      ),
+                      const SizedBox(width: 10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart:106]: same

File: Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart [hunk 38]
```diff
@@ -0,0 +1,227 @@
...
+                      SizedBox(
+                        height: 20.0,
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart:108]: maybe you can put the values in style file

File: Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart [hunk 39]
```diff
@@ -0,0 +1,227 @@
...
+                            return null;
+                        }
+                      }
+
+                      return '* Required field.';
+                    },
+                    onSaved: (value) => inputCode = value ?? '',
+                  ),
+                ),
+                const SizedBox(height: 20.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/password-recuperation/password_recuperation.dart:152]: Add in global style file

File: Configurateur/front/lib/screens/register-confirmation/confirmed_user.dart [hunk 40]
```diff
@@ -53,44 +55,58 @@ class ConfirmedUserState extends State<ConfirmedUser> {
...
+              Text(
+                "Votre inscription a bien été confirmée, vous pouvez maintenant vous connecter et profiter de notre application",
+                style: TextStyle(
+                  fontSize: screenFormat == ScreenFormat.desktop
+                      ? desktopBigFontSize
+                      : tabletBigFontSize,
+                ),
+                textAlign: TextAlign.center,
+              ),
+              SizedBox(
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/register-confirmation/confirmed_user.dart:81]: Add in global style file

File: Configurateur/front/lib/screens/register-confirmation/register_confirmation.dart [hunk 41]
```diff
@@ -126,11 +143,15 @@ class RegisterConfirmationState extends State<RegisterConfirmation> {
                         padding: const EdgeInsets.all(10.0),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/register-confirmation/register_confirmation.dart:143]: Add in global style file

File: Configurateur/front/lib/screens/save_container/my_container.dart [hunk 42]
```diff
@@ -70,14 +74,18 @@ class MyContainerState extends State<MyContainer> {
...
+            Text(
               "Mes conteneurs sauvegardés",
-              style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
+              style: TextStyle(
+                  fontSize: screenFormat == ScreenFormat.desktop
+                      ? desktopFontSize
+                      : tabletFontSize,
+                  fontWeight: FontWeight.bold),
             ),
             const SizedBox(height: 20),
```
Comment (Louismaestre) [inline Configurateur/front/lib/screens/save_container/my_container.dart:85]: Add in global style file

File: Configurateur/front/pubspec.lock
Comment (Louismaestre) [inline Configurateur/front/pubspec.lock:1]: I think it's a mistake

**Review by H-chauvet — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]
LGTM

**Pull request merged**